From 4d4fbc7c91d7b843165e29f2343e359d02459b59 Mon Sep 17 00:00:00 2001
From: "Chen, ZhiminX" <zhiminx.chen@intel.com>
Date: Tue, 9 Jul 2019 18:00:16 +0800
Subject: [PATCH 1142/1214] ablbc: fix node permission and reboot notifier call
 for capsule update

1. To know if capsule update is needed, the system app owned by non-root
   needs the read permission to /sys/kernel/capsule/capsule_requested,
   so set default mode 0444 to allow non-root access.
2. When capsule update is requested and the name id of reboot target is 0,
   the nvram is used to store capsule message. In this case should not
   process writing target message to nvram in set_reboot_target(), avoid
   message about capsule update in nvram being overwritten.

Change-Id: Ife2a276fdd40280500fa00f86d26a16f5319a2f5
Signed-off-by: Chen, ZhiminX <zhiminx.chen@intel.com>
Tracked-On: OAM-83554
---
 drivers/staging/android/abl/ablbc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/android/abl/ablbc.c b/drivers/staging/android/abl/ablbc.c
index 78bfd02..e58b4f8 100644
--- a/drivers/staging/android/abl/ablbc.c
+++ b/drivers/staging/android/abl/ablbc.c
@@ -239,7 +239,7 @@ static struct kobj_attribute capsule_name_attribute =
 		__ATTR(capsule_name, 0600, NULL, capsule_store);
 
 static struct kobj_attribute capsule_requested_attribute =
-		__ATTR(capsule_requested, 0400, is_capsule_requested, NULL);
+		__ATTR(capsule_requested, 0444, is_capsule_requested, NULL);
 
 static int reboot_target_name2id(const char *name)
 {
@@ -270,6 +270,8 @@ static int set_reboot_target(const char *name)
 		       __func__, name);
 		return -EINVAL;
 	}
+	if ((id == 0) && capsule_request)
+		return 0;
 
 	cdh.data = 0;
 	cdh.length = 2; /* 2*32 bits, from header to padding */
-- 
2.7.4

